src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs

```diff
@@ -290,11 +291,42 @@ public Blend Blend
             }
             set
             {
-                // Allocate temporary native memory buffer
-                // and copy input blend factors into it.
+                if (value == null || value.Factors == null)
+                {
+                    // This is the behavior on Desktop
+                    throw new NullReferenceException();
+                }
+
+                if (value.Positions == null)
+                {
+                    throw new ArgumentNullException("source");
```

Comment:
No, it's fine to keep the behavior, it just looks strange at first. Maybe just add a comment summarizing your second paragraph.

We could probably (compatibly) change the string parameter if we really wanted to, but in general I don't think it's worth it for this library. I'd rather everything just be identical to desktop for simplicity.